PWGHF/Core/HfMlResponse.h

```diff
@@ -62,14 +64,16 @@ uint8_t tagPID(const T1& track)
 template <typename T1, typename T2>
 T1 getCombinedNSigma(const T1& nSigTpc, const T1& nSigTof, const T2& tagPID)
 {
+  float combinedNSigma{0.};
   if (TESTBIT(tagPID, PIDStatus::TpcAndTof)) {
-    return std::sqrt((nSigTpc * nSigTpc + nSigTof * nSigTof) / 2);
-  }
-  if (TESTBIT(tagPID, PIDStatus::TpcOnly)) {
-    return std::abs(nSigTpc);
+    combinedNSigma = std::sqrt((nSigTpc * nSigTpc + nSigTof * nSigTof) / 2);
+  } else if (TESTBIT(tagPID, PIDStatus::TpcOnly)) {
+    combinedNSigma = std::abs(nSigTpc);
+  } else if (TESTBIT(tagPID, PIDStatus::TofOnly)) {
+    combinedNSigma = std::abs(nSigTof);
   }
 
-  return std::abs(nSigTof); // if (TESTBIT(tagPID, PIDStatus::TofOnly))
+  return combinedNSigma;
```

Comment:
Why did you change the logic from returning the value right away to assigning it to a local variable?

Reply:
Because I get a Warning at compilation time if I only have `return` inside `if loops`.

Reply:
You can add `return 0.;` after the last `if`.

Reply:
Okay, thanks :)